website/docs/chapter-04/01_setups.md

```diff
@@ -0,0 +1,180 @@
+```
+
+
+
+## 4.1.2 Test|Task Setup
+
+::::lo[Learning Objectives]
+
+:::K1[LO-4.1.2-1]
+
+Recall key characteristics, benefits, and syntax of Test Setup
+
+:::
+
+:::K2[LO-4.1.2-2]
+
+Understand when Test|Task Setup is executed and used
+
+:::
+
+::::
+
+A **Test|Task Setup** is executed before a single test|task runs.
+It is used to prepare the specific conditions required for that test|task.
+
+You can define a default Test|Task Setup in the `*** Settings ***` section of the suite using the `Test Setup`|`Task Setup` setting.
+This setup will be applied to all tests|tasks within the suite unless overridden.
+
+Individual tests|tasks can override the default setup by specifying their own `[Setup]` setting within the test|task.
+To disable the setup for a specific test|task, you can set `[Setup]    NONE`, which means that no setup will be executed for that test|task.
+
+**Key characteristics of Test|Task Setup:**
+- Test|Task Setup is a single keyword call with potential argument values.
+- Executed before the test|task starts.
+- If the Test|Task Setup fails, the test|task is marked as failed, and its body, including its main keywords, is not executed.
+- Can be set globally for all tests|tasks in a suite and overridden locally.
+- Logged in the execution log as a separate section, indicating the setup status.
+
+**Typical use cases:**
+- Setting up data unique to the test|task.
+- Executing preparation steps to navigate to the automated task or feature under test.
+- Distinguishing phases of a test|task in *setup* (aka *preparation* or *precondition checking*), *steps*, and *teardown* (aka *clean up* or *postconditions*).
+
+Example of defining a default Test|Task Setup in the suite settings and overriding it on a test case:
+
+```robotframework
+*** Settings ***
+Test Setup    Login As Standard User
+
+
+*** Test Cases ***
+User Action Test With Default Setup    # Default Test Setup is applied
+    Perform User Actions    0815
+
+Another User Action With Default Setup    # Default Test Setup is applied
+    Perform another User Action    4711
+
+Admin Access Test With Local Setup
+    [Setup]    Login As Admin    # Override the default setup
+    Perform Admin Actions   007
+
+No Setup Test
+    [Setup]    NONE    # Override and disable the setup by case-sensitive NONE
+    Perform Actions Without Login   000
+```
+
+
+
+## 4.1.3 Keyword Setup
+
+::::lo[Learning Objectives]
+
+:::K1[LO-4.1.3]
+
+Recall key characteristics and syntax of Keyword Setup
+
+:::
+
+::::
+
+A **Keyword Setup** is executed before the body of a user keyword is executed.
+It allows for preparation steps specific to that keyword or ensures that the keyword's requirements are met before execution.
+
+**Key characteristics of Keyword Setup:**
+- Keyword Setup is a single keyword call with potential argument values.
+- Executed before the keyword's body.
+- If the Keyword Setup fails, the keyword's body is not executed.
```

Comment:
- If the Keyword Setup fails, the test|task is marked as failed, and the keyword's body is not executed.

Reply:
not necessarily.
the keyword would be fail, but not the test|task.

let's assume, the failing keyword is in a try block.